### ISSUE TITLE
localhost with target microbit give Not found :(

### ISSUE THREAD

**Issue by OwenBrotherwood** [Microsoft/pxt#566]
Using doc https://www.pxt.io/cli and a clone of repro @ https://github.com/Microsoft/pxt/commit/c7c96fe77c058956cfa97e43c7ab8d7139b9b0c2 

pxt target microbit
pxt serve 

results in message to open URL
http://localhost:3232/#local_token=8cbb562f-970c-40b5-155e-87e76bccd3e3

Web page gives
Not found :(

```
pxt target microbit
Installing pxt-microbit locally; don't worry about package.json warnings.
\
> serialport@4.0.3 install /mnt/c/Users/owen/Documents/GitHub/m/node_modules/pxt-microbit/node_modules/pxt-core/node_modules/serialport
> node-pre-gyp install --fallback-to-build

[serialport] Success: "/mnt/c/Users/owen/Documents/GitHub/m/node_modules/pxt-microbit/node_modules/pxt-core/node_modules/serialport/build/Release/serialport.node" is installed via remote
pxt-microbit@0.5.1 node_modules/pxt-microbit
└── pxt-core@0.5.1 (marked@0.3.6, lzma@2.3.2, bluebird@3.4.6, faye-websocket@0.11.0, highlight.js@9.7.0, serialport@4.0.3)
Installed PXT/microbit. To start server run:
    pxt serve

pxt serve
Using PXT/microbit from /mnt/c/Users/owen/Documents/GitHub/m/node_modules/pxt-microbit.
Going to /mnt/c/Users/owen/Documents/GitHub/m/node_modules/pxt-microbit
No sim/tsconfig.json; assuming npm installed package
Starting server in /mnt/c/Users/owen/Documents/GitHub/m/node_modules/pxt-microbit
With pxt core at /mnt/c/Users/owen/Documents/GitHub/m/node_modules/pxt-core
serial: monitoring ports...
starting local ws server at 3233...
---------------------------------------------

To launch the editor, open this URL:
http://localhost:3232/#local_token=8cbb562f-970c-40b5-155e-87e76bccd3e3

---------------------------------------------
opening http://localhost:3232/#local_token=8cbb562f-970c-40b5-155e-87e76bccd3e3
^C
uname -a # Microsoft Canonical (Ubuntu) Linux
Linux SF100WS005521 3.4.0+ #1 PREEMPT Thu Aug 1 17:06:05 CST 2013 x86_64 x86_64 x86_64 GNU/Linux
node -version
v4.6.1
npm --version
2.15.9 
```

**Comment by OwenBrotherwood** [comment]
The same result when using the "npm code" ie not building locally.
npm install -g ptx
ptx target microbit
ptx serve

**Comment by OwenBrotherwood** [comment]
ptx is pt. not available:
```
 npm install -g ptx
npm ERR! Linux 3.4.0+
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install" "-g" "ptx"
npm ERR! node v4.6.1
npm ERR! npm  v2.15.9
npm ERR! code E404

npm ERR! 404 Registry returned 404 for GET on https://registry.npmjs.org/ptx
npm ERR! 404
npm ERR! 404 'ptx' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
npm ERR! 404
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.

npm ERR! Please include the following file with any support request:
npm ERR!     /mnt/c/Users/owen/Documents/GitHub/m/npm-debug.log
```

**Comment by OwenBrotherwood** [comment]
Same result from a standard Windows 10 cmd prompt: Not found 

```
C:\Users\owen>npm install -g pxt
C:\Users\owen\AppData\Roaming\npm\pxt -> C:\Users\owen\AppData\Roaming\npm\node_modules\pxt\pxt
pxt@0.3.0 C:\Users\owen\AppData\Roaming\npm\node_modules\pxt

C:\Users\owen>cd mb

C:\Users\owen\mb>pxt target microbit
Installing pxt-microbit locally; don't worry about package.json warnings.
\
> serialport@4.0.3 install C:\Users\owen\mb\node_modules\pxt-microbit\node_modules\pxt-core\node_modules\serialport
> node-pre-gyp install --fallback-to-build

[serialport] Success: "C:\Users\owen\mb\node_modules\pxt-microbit\node_modules\pxt-core\node_modules\serialport\build\Release\serialport.node" is installed via remote
pxt-microbit@0.5.7 node_modules\pxt-microbit
├── semantic-ui-less@2.2.4 (jquery@3.1.1)
├── less@2.7.1 (graceful-fs@4.1.9, mime@1.3.4, image-size@0.5.0, errno@0.1.4, source-map@0.5.6, mkdirp@0.5.1, promise@7.1.1)
└── pxt-core@0.5.9 (marked@0.3.6, lzma@2.3.2, bluebird@3.4.6, faye-websocket@0.11.0, highlight.js@9.7.0, serialport@4.0.3)
Installed PXT/microbit. To start server run:
    pxt serve

C:\Users\owen\mb>pxt serve
Using PXT/microbit from C:\Users\owen\mb\node_modules\pxt-microbit.
Going to C:\Users\owen\mb\node_modules\pxt-microbit
No sim/tsconfig.json; assuming npm installed package
Starting server in C:\Users\owen\mb\node_modules\pxt-microbit
With pxt core at C:\Users\owen\mb\node_modules\pxt-core
serial: monitoring ports...
starting local ws server at 3233...
---------------------------------------------

To launch the editor, open this URL:
http://localhost:3232/#local_token=55bc0701-c110-4da7-6e9b-37ca0f188ce7

---------------------------------------------
opening http://localhost:3232/#local_token=55bc0701-c110-4da7-6e9b-37ca0f188ce7

```
C:\Users\owen\mb>node --version
v4.6.1

C:\Users\owen\mb>npm --version
2.15.9

**Comment by OwenBrotherwood** [comment]
Future test area with redhat @ http://pxt-owenbrotherwood.rhcloud.com/
Let us see if I get time.

Any thoughts on what I should check?

**Comment by pelikhan** [comment]
Could you try upgrade node.js to 5+?

**Comment by OwenBrotherwood** [comment]
That did the job
Off to test on Windows Canonical (ubuntu) Linux
Thanks: did I oversee a requirement for node version :( ?

```
Microsoft Windows [Version 10.0.14393]
(c) 2016 Microsoft Corporation. Alle rettigheder forbeholdes.

C:\Users\owen>npm --version
3.10.8

C:\Users\owen>node --version
v6.9.1

C:\Users\owen>npm uninstall -g pct
Terminate batch job (Y/N)? npm uninstall -g pct
^C
C:\Users\owen>

C:\Users\owen>npm uninstall -g pxt
- pxt@0.3.0 node_modules\pxt

C:\Users\owen>rmdir /s mb
mb, Are you sure (Y/N)? y

C:\Users\owen>mkdir mb

C:\Users\owen>cd mb

C:\Users\owen\mb>npm install -g pxt
C:\Users\owen\AppData\Roaming\npm\pxt -> C:\Users\owen\AppData\Roaming\npm\node_modules\pxt\pxt
C:\Users\owen\AppData\Roaming\npm
`-- pxt@0.3.0


C:\Users\owen\mb>ls
'ls' is not recognized as an internal or external command,
operable program or batch file.

C:\Users\owen\mb>dir
 Volume in drive C has no label.
 Volume Serial Number is 0EE0-492B

 Directory of C:\Users\owen\mb

25-10-2016  20:48    <DIR>          .
25-10-2016  20:48    <DIR>          ..
               0 File(s)              0 bytes
               2 Dir(s)  88.322.547.712 bytes free

C:\Users\owen\mb>pxt target microbit
Installing pxt-microbit locally; don't worry about package.json warnings.

> serialport@4.0.3 install C:\Users\owen\mb\node_modules\serialport
> node-pre-gyp install --fallback-to-build

[serialport] Success: "C:\Users\owen\mb\node_modules\serialport\build\Release\serialport.node" is installed via remote
C:\Users\owen\mb
`-- pxt-microbit@0.5.9
  +-- less@2.7.1
  | +-- errno@0.1.4
  | | `-- prr@0.0.0
  | +-- graceful-fs@4.1.9
  | +-- image-size@0.5.0
  | +-- mime@1.3.4
  | +-- mkdirp@0.5.1
  | | `-- minimist@0.0.8
  | +-- promise@7.1.1
  | | `-- asap@2.0.5
  | `-- source-map@0.5.6
  +-- pxt-core@0.5.11
  | +-- bluebird@3.4.6
  | +-- faye-websocket@0.11.0
  | | `-- websocket-driver@0.6.5
  | |   `-- websocket-extensions@0.1.1
  | +-- highlight.js@9.7.0
  | +-- lzma@2.3.2
  | +-- marked@0.3.6
  | `-- serialport@4.0.3
  |   +-- bindings@1.2.1
  |   +-- commander@2.9.0
  |   | `-- graceful-readlink@1.0.1
  |   +-- debug@2.2.0
  |   | `-- ms@0.7.1
  |   +-- lie@3.1.0
  |   | `-- immediate@3.0.6
  |   +-- nan@2.4.0
  |   +-- node-pre-gyp@0.6.30
  |   | +-- mkdirp@0.5.1
  |   | | `-- minimist@0.0.8
  |   | +-- nopt@3.0.6
  |   | | `-- abbrev@1.0.9
  |   | +-- npmlog@4.0.0
  |   | | +-- are-we-there-yet@1.1.2
  |   | | | +-- delegates@1.0.0
  |   | | | `-- readable-stream@2.1.5
  |   | | |   +-- buffer-shims@1.0.0
  |   | | |   +-- core-util-is@1.0.2
  |   | | |   +-- inherits@2.0.3
  |   | | |   +-- isarray@1.0.0
  |   | | |   +-- process-nextick-args@1.0.7
  |   | | |   +-- string_decoder@0.10.31
  |   | | |   `-- util-deprecate@1.0.2
  |   | | +-- console-control-strings@1.1.0
  |   | | +-- gauge@2.6.0
  |   | | | +-- aproba@1.0.4
  |   | | | +-- has-color@0.1.7
  |   | | | +-- has-unicode@2.0.1
  |   | | | +-- object-assign@4.1.0
  |   | | | +-- signal-exit@3.0.1
  |   | | | +-- string-width@1.0.2
  |   | | | | +-- code-point-at@1.0.1
  |   | | | | | `-- number-is-nan@1.0.1
  |   | | | | `-- is-fullwidth-code-point@1.0.0
  |   | | | |   `-- number-is-nan@1.0.1
  |   | | | +-- strip-ansi@3.0.1
  |   | | | | `-- ansi-regex@2.0.0
  |   | | | `-- wide-align@1.1.0
  |   | | `-- set-blocking@2.0.0
  |   | +-- rc@1.1.6
  |   | | +-- deep-extend@0.4.1
  |   | | +-- ini@1.3.4
  |   | | +-- minimist@1.2.0
  |   | | `-- strip-json-comments@1.0.4
  |   | +-- request@2.75.0
  |   | | +-- aws-sign2@0.6.0
  |   | | +-- aws4@1.4.1
  |   | | +-- bl@1.1.2
  |   | | | `-- readable-stream@2.0.6
  |   | | |   +-- core-util-is@1.0.2
  |   | | |   +-- inherits@2.0.3
  |   | | |   +-- isarray@1.0.0
  |   | | |   +-- process-nextick-args@1.0.7
  |   | | |   +-- string_decoder@0.10.31
  |   | | |   `-- util-deprecate@1.0.2
  |   | | +-- caseless@0.11.0
  |   | | +-- combined-stream@1.0.5
  |   | | | `-- delayed-stream@1.0.0
  |   | | +-- extend@3.0.0
  |   | | +-- forever-agent@0.6.1
  |   | | +-- form-data@2.0.0
  |   | | | `-- asynckit@0.4.0
  |   | | +-- har-validator@2.0.6
  |   | | | +-- chalk@1.1.3
  |   | | | | +-- ansi-styles@2.2.1
  |   | | | | +-- escape-string-regexp@1.0.5
  |   | | | | +-- has-ansi@2.0.0
  |   | | | | | `-- ansi-regex@2.0.0
  |   | | | | +-- strip-ansi@3.0.1
  |   | | | | | `-- ansi-regex@2.0.0
  |   | | | | `-- supports-color@2.0.0
  |   | | | +-- is-my-json-valid@2.15.0
  |   | | | | +-- generate-function@2.0.0
  |   | | | | +-- generate-object-property@1.2.0
  |   | | | | | `-- is-property@1.0.2
  |   | | | | +-- jsonpointer@4.0.0
  |   | | | | `-- xtend@4.0.1
  |   | | | `-- pinkie-promise@2.0.1
  |   | | |   `-- pinkie@2.0.4
  |   | | +-- hawk@3.1.3
  |   | | | +-- boom@2.10.1
  |   | | | +-- cryptiles@2.0.5
  |   | | | +-- hoek@2.16.3
  |   | | | `-- sntp@1.0.9
  |   | | +-- http-signature@1.1.1
  |   | | | +-- assert-plus@0.2.0
  |   | | | +-- jsprim@1.3.1
  |   | | | | +-- extsprintf@1.0.2
  |   | | | | +-- json-schema@0.2.3
  |   | | | | `-- verror@1.3.6
  |   | | | `-- sshpk@1.10.1
  |   | | |   +-- asn1@0.2.3
  |   | | |   +-- assert-plus@1.0.0
  |   | | |   +-- bcrypt-pbkdf@1.0.0
  |   | | |   +-- dashdash@1.14.0
  |   | | |   +-- ecc-jsbn@0.1.1
  |   | | |   +-- getpass@0.1.6
  |   | | |   +-- jodid25519@1.0.2
  |   | | |   +-- jsbn@0.1.0
  |   | | |   `-- tweetnacl@0.14.3
  |   | | +-- is-typedarray@1.0.0
  |   | | +-- isstream@0.1.2
  |   | | +-- json-stringify-safe@5.0.1
  |   | | +-- mime-types@2.1.12
  |   | | | `-- mime-db@1.24.0
  |   | | +-- node-uuid@1.4.7
  |   | | +-- oauth-sign@0.8.2
  |   | | +-- qs@6.2.1
  |   | | +-- stringstream@0.0.5
  |   | | +-- tough-cookie@2.3.1
  |   | | `-- tunnel-agent@0.4.3
  |   | +-- rimraf@2.5.4
  |   | | `-- glob@7.1.0
  |   | |   +-- fs.realpath@1.0.0
  |   | |   +-- inflight@1.0.5
  |   | |   | `-- wrappy@1.0.2
  |   | |   +-- inherits@2.0.3
  |   | |   +-- minimatch@3.0.3
  |   | |   | `-- brace-expansion@1.1.6
  |   | |   |   +-- balanced-match@0.4.2
  |   | |   |   `-- concat-map@0.0.1
  |   | |   +-- once@1.4.0
  |   | |   | `-- wrappy@1.0.2
  |   | |   `-- path-is-absolute@1.0.1
  |   | +-- semver@5.3.0
  |   | +-- tar@2.2.1
  |   | | +-- block-stream@0.0.9
  |   | | +-- fstream@1.0.10
  |   | | | `-- graceful-fs@4.1.9
  |   | | `-- inherits@2.0.3
  |   | `-- tar-pack@3.1.4
  |   |   +-- fstream@1.0.10
  |   |   | +-- graceful-fs@4.1.9
  |   |   | `-- inherits@2.0.3
  |   |   +-- fstream-ignore@1.0.5
  |   |   | +-- inherits@2.0.3
  |   |   | `-- minimatch@3.0.3
  |   |   |   `-- brace-expansion@1.1.6
  |   |   |     +-- balanced-match@0.4.2
  |   |   |     `-- concat-map@0.0.1
  |   |   +-- once@1.3.3
  |   |   | `-- wrappy@1.0.2
  |   |   +-- readable-stream@2.1.5
  |   |   | +-- buffer-shims@1.0.0
  |   |   | +-- core-util-is@1.0.2
  |   |   | +-- inherits@2.0.3
  |   |   | +-- isarray@1.0.0
  |   |   | +-- process-nextick-args@1.0.7
  |   |   | +-- string_decoder@0.10.31
  |   |   | `-- util-deprecate@1.0.2
  |   |   `-- uid-number@0.0.6
  |   `-- object.assign@4.0.4
  |     +-- define-properties@1.1.2
  |     | `-- foreach@2.0.5
  |     +-- function-bind@1.1.0
  |     `-- object-keys@1.0.11
  `-- semantic-ui-less@2.2.4
    `-- jquery@3.1.1

npm WARN enoent ENOENT: no such file or directory, open 'C:\Users\owen\mb\package.json'
npm WARN mb No description
npm WARN mb No repository field.
npm WARN mb No README data
npm WARN mb No license field.
Installed PXT/microbit. To start server run:
    pxt serve

C:\Users\owen\mb>dir
 Volume in drive C has no label.
 Volume Serial Number is 0EE0-492B

 Directory of C:\Users\owen\mb

25-10-2016  20:49    <DIR>          .
25-10-2016  20:49    <DIR>          ..
25-10-2016  20:49    <DIR>          node_modules
               0 File(s)              0 bytes
               3 Dir(s)  88.238.002.176 bytes free

C:\Users\owen\mb>pxt serve
Using PXT/microbit from C:\Users\owen\mb\node_modules\pxt-microbit.
Going to C:\Users\owen\mb\node_modules\pxt-microbit
No sim/tsconfig.json; assuming npm installed package
Starting server in C:\Users\owen\mb\node_modules\pxt-microbit
With pxt core at C:\Users\owen\mb\node_modules\pxt-core
serial: monitoring ports...
starting local ws server at 3233...
---------------------------------------------

To launch the editor, open this URL:
http://localhost:3232/#local_token=55bc0701-c110-4da7-6e9b-37ca0f188ce7

---------------------------------------------
opening http://localhost:3232/#local_token=55bc0701-c110-4da7-6e9b-37ca0f188ce7
ws connection at /55bc0701-c110-4da7-6e9b-37ca0f188ce7/serial
```

**Comment by OwenBrotherwood** [comment]
<img width="960" alt="localhost" src="https://cloud.githubusercontent.com/assets/866122/19699770/e893244c-9af4-11e6-8c97-2c2c9c1eed0a.PNG">

**Comment by pelikhan** [comment]
That's a bug on our side. We should specify the minimum node/npm version in the package.

**Comment by OwenBrotherwood** [comment]
OK: "Not found" has raised it's head in a pure Windows 10 installation while fixing the pxt-microbit-electron app.
I am pretty sure the problem is in pxt code: I keep looking
<img width="476" alt="windows-10-not-found" src="https://cloud.githubusercontent.com/assets/866122/19835127/03ed192e-9e7d-11e6-9a7e-8fd13238e1f8.PNG">

**Comment by OwenBrotherwood** [comment]
Something nasty may be happening around this function:
```
function getPxtTarget() {
    if (fs.existsSync(exports.targetDir + "/built/target.json")) {
        var res = readJson(exports.targetDir + "/built/target.json");
        if (res.id && res.bundledpkgs)
            return res;
    }
    var raw = readJson(exports.targetDir + "/pxtarget.json");
    raw.bundledpkgs = {};
    return raw;
}
```

**Comment by pelikhan** [comment]
Minimum version specified in https://github.com/Microsoft/pxt/pull/706

**Comment by OwenBrotherwood** [comment]
I have commented https://github.com/Microsoft/pxt/pull/706#issuecomment-258788376

**Comment by abchatra** [comment]
Fixed #706
